src/mapping/Partition.hpp

```diff
@@ -0,0 +1,315 @@
+   */
+  Partition(int               dimension,
+            mesh::Vertex      center,
+            double            radius,
+            double            parameter,
+            std::vector<bool> deadAxis,
+            Polynomial        polynomial,
+            unsigned int      targetSize,
+            mesh::PtrMesh     inputMesh,
+            mesh::PtrMesh     outputMesh);
+
+  /// Removes a computed mapping.
+  void clear();
+
+  /// Execute a consistent mapping
+  void mapConsistent(mesh::PtrData inputData, mesh::PtrData outputData);
+
+  /// Execute a consistent mapping
+  void mapConservative(mesh::PtrData inputData, mesh::PtrData outputData);
+
+  /// @brief set the normalized weight in the normalizedWeight data structure
+  void setNormalizedWeight(double normalizedWeight, VertexID vertexID);
+
+  /// Compute the weight for a given vertex
+  double computeWeight(const mesh::Vertex &v) const;
+
+  /// Evaluate the interpolant for a given vertex
+  double getInterpolatedValue(const VertexID id, unsigned int direction) const;
+
+  /// Indicate whether a vertex lies within the partition or not
+  bool isVertexInside(const mesh::Vertex &v) const;
+
+  /// Indicate that the mapped data is now considered as outdated by setting _hasMappedData to false
+  void setMappingFinished();
+
+  /// Number of input vertices this partition operates on
+  unsigned int getNumberOfInputVertices() const;
+
+  /// The center coordinate of this partition
+  std::array<double, 3> getCenterCoords() const;
+
+  /// Indicates whether there are valid output points for this partition
+  bool isEmpty() const;
+
+private:
+  precice::logging::Logger _log{"mapping::Partition"};
+
+  bool _hasComputedMapping = false;
+  bool _hasMappedData      = false;
+  bool _emptyPartition     = false;
+
+  /// Partition center
+  mesh::Vertex _center;
+
+  /// Partition radius
+  const double _radius;
+
+  /// Array with one entry per data component. A plain array would require knowledge about the number of components
+  std::array<Eigen::VectorXd, 3> resultVector;
+
+  /// vector containing the normalized weights (only for conservative mappings)
+  Eigen::VectorXd normalizedWeights;
+
+  // Stores the global IDs of the vertices so that we can apply a binary
+  // search in order to query specific objects
+  boost::container::flat_set<VertexID> inputIDs;
+  boost::container::flat_set<VertexID> outputIDs;
+
+  /// The RBF solver
+  RadialBasisFctSolver<RADIAL_BASIS_FUNCTION_T> _rbfSolver;
+
+  /// Polynomial treatment in the RBF solver
+  Polynomial _polynomial;
+
+  //TODO: Can probably be removed
+  RADIAL_BASIS_FUNCTION_T _basisFunction;
+
+  /// The weighting function
+  CompactPolynomialC2 _weightingFunction;
+};
+
+// --------------------------------------------------- HEADER IMPLEMENTATIONS
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+Partition<RADIAL_BASIS_FUNCTION_T>::Partition(
+    int               dimension,
+    mesh::Vertex      center,
```

Comment:
What if the center is tagged/untagged before the construction?
Aren't there alternatives to using copies of `mesh::Vertex` inside the Partition?

Reply:
Tagging has no influence on the class construction.
Similar to other comments here: I mainly used a `Vertex`, because that's what the set of coordinates represent. I can use raw coordinates, i.e., `Eigen::Vector3d` or something similar, but it's less expressive in my opinion. What are your concerns regarding the vertex class here?

Reply:
The point I am trying to make is that there is some state attached to the Vertex.
If this doesn't affect the construction, then it's fine.